doc-examples/jdbc-spring-template-example-groovy/src/main/groovy/example/AbstractBookRepository.groovy

```diff
@@ -0,0 +1,33 @@
+package example
+
+import io.micronaut.context.annotation.Requires
+import io.micronaut.core.annotation.NonNull
+import io.micronaut.data.connection.jdbc.advice.DelegatingDataSource
+import io.micronaut.data.jdbc.annotation.JdbcRepository
+import io.micronaut.data.model.query.builder.sql.Dialect
+import io.micronaut.data.repository.CrudRepository
+import jakarta.transaction.Transactional
+import jakarta.validation.Valid
+import jakarta.validation.constraints.NotNull
+import org.springframework.jdbc.core.JdbcTemplate
+
+import javax.sql.DataSource
+
+@Requires(property = 'spec.name', value = 'BookRepositorySpec')
+// tag::clazz[]
+@JdbcRepository(dialect = Dialect.H2)
+abstract class AbstractBookRepository implements CrudRepository<@Valid Book, @NotNull Long> {
+
+    private final JdbcTemplate jdbcTemplate; //<2>
```

Comment:
can the JdbcTemplate be injected directly without the need of instantiating it manually?

Reply:
No, that didn't work for me, and even in that case it did, it still needs a DataSource set before you can use it. I patterned this after what is done in: https://github.com/micronaut-projects/micronaut-data/blob/master/data-spring-jdbc/src/main/java/io/micronaut/data/spring/jdbc/SpringJdbcConnectionOperations.java

Reply:
I suppose we could create a factory that does the same thing, and then it could be injected.